### PR TITLE
ISPN-13690 Feature form for secured caches

### PR DESCRIPTION
**Pull request by dipguptaredhat**
Feature secured cache added
https://issues.redhat.com/browse/ISPN-13690

### CODE REVIEW

File: src/services/cacheService.ts [hunk 1]
```diff
@@ -621,4 +621,15 @@ export class CacheService {
       errorMessage: `Cache ${cacheName} already exists.`,
     });
   }
+
+  /**
+   * Retrieve security roles
+   *
+   */
+  public async getSecurityRoles(): Promise<Either<ActionResponse, string[]>> {
```
Comment (karesti) [inline src/services/cacheService.ts:629]: move to securityService
this method is unrelated to caches

File: src/app/Caches/Create/Features/SecuredCache.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    const onSelectRoles = (event, selection) => {
+        if (roles.includes(selection))
+            setRoles(roles.filter(role => role !== selection));
+        else
+            setRoles([...roles, selection]);
+    }
+
+    return (
+        <Form>
```
Comment (karesti) [inline src/app/Caches/Create/Features/SecuredCache.tsx:72]: check the Card and CardBody changes I reported in indexing to change this accordingly.

File: src/app/Caches/Create/Features/SecuredCache.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        }
+    }, [loading]);
+
+    useEffect(() => {
+        props.securedOptionsModifier({
+            roles: roles,
+        });
+    }, [roles]);
+
+    // Options for Isolation Level
```
Comment (karesti) [inline src/app/Caches/Create/Features/SecuredCache.tsx:56]: delete comment

File: src/app/Caches/Create/Features/SecuredCache.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+const SecuredCache = (props: {
+    securedOptions: SecuredCache,
+    securedOptionsModifier: (SecuredCache) => void,
+}) => {
+
+    const { t } = useTranslation();
+    const brandname = t('brandname.brandname');
+
+    //Secured Cache
```
Comment (karesti) [inline src/app/Caches/Create/Features/SecuredCache.tsx:25]: delete this comment, not necessary information

**Review by karesti — CHANGES_REQUESTED** [review]
some changes that depend from the changes I added / asked in the indexing PR

**Comment by dipguptaredhat** [comment]
@karesti new commit added, in the last step create cache not working, please check

**Comment by karesti** [comment]
@dipguptaredhat rebase needed from main! ping me DM when ready please!

**Comment by karesti** [comment]
@dipguptaredhat choosing a role should be mandatory for secured caches next. otherwise we are at the end with a configuration that is not correct

```json
 "security": {
      "authorization": {
        "enabled": true,
        "roles": []
      }
    }
```

**Comment by karesti** [comment]
@dipguptaredhat there was a bug in 13.0.10.Final and 14.x parsing the configuration of secured caches. works with 13.0.10.Final but it's not fixed in 14.0.0.Dev02
https://github.com/infinispan/infinispan/commit/82c1e00a2c7c6b8cc9e4950b9efde13eac5712a6

**Comment by oraNod** [comment]
@karesti and @dipguptaredhat Hey, what do you think about changing this to "RBAC" instead of "secured"?

When I think of security I think of SSL/TLS so changing the label to RBAC might be more immediately descriptive to users. They'd know what to expect when they select the option.

I ask this because I can change the labels but they won't necessarily correspond to Infinispan types, etc. So I would propose that if we make this change we do a bit of refactoring to s/secured/rbac. I know that's a pain though. Sorry I didn't mention it earlier but I've been caught up in stuff.

**Comment by tristantarrant** [comment]
I'm not sure the RBAC acronym is well-known enough. I'd think that ACL is much more common.

**Comment by oraNod** [comment]
Maybe we could even go with "authorization" to be consistent with the schema?

**Comment by karesti** [comment]
@oraNod make any doc/label change you need. I vote for authorization too

**Review by karesti — CHANGES_REQUESTED** [review]
Rename header label for Authorization instead of secured

Chose at least 1 role before moving forward.

**Comment by dipguptaredhat** [comment]
@karesti changes done and squashed

**Review by karesti — APPROVED** [review]

**Pull request merged**